### PR TITLE
search part done

### PR DESCRIPTION
**Pull request by Rehaan1**
# Pull Request Type
Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

# Purpose
- WHAT is this PR for?
addition of new feature
# Approach
- HOW does this change fulfill the purpose?
using getsongkey api endopoint gets the required data of key and tempo
# Does this PR introduce any new bug?
- [ ] Yes
- [X] No
- [ ] Maybe

# Other information

# Did you learn something new?
- Describe what did you learn
api calls and waiting for data to come before processing further

### CODE REVIEW

**Comment by BRO3886** [comment]
`https://api.getsongbpm.com/song/?api_key=` is being used on every route almost, can you replace this with a var like `GET_BPM_BASE_URL`

@rithikjain go through the code once

File: api/routes/search.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
+const express = require('express')
+const https = require('https')
+const router = express.Router()
+
+router.post('/song_and_artist', (req, res) => {
```
Comment (rithikjain) [inline api/routes/search.js]: Route name in snake_case is not a convention, try to go with camelCase

File: api/routes/search.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+  if (!req.body.artist) {
+    return res.status(400).json({
+      error: 'missing required parameters. refer documentation'
+    })
+  }
+
+  const songName = req.body.song_name.split(' ').join('+')
+  const artist = req.body.artist.split(' ').join('+')
+  const url = 'https://api.getsongbpm.com/search/?api_key=' + process.env.API_KEY + '&type=both&lookup=song:' + songName + ' artist:' + artist
```
Comment (rithikjain) [inline api/routes/search.js]: Make a variable for the base url as it will be used multiple times

**Review by rithikjain — CHANGES_REQUESTED** [review]
Please fix the things I have suggested changes for.

File: api/routes/search.js (hunk 1 above)
Comment (Rehaan1) [reply api/routes/search.js]: Will make the changes

File: api/routes/search.js (hunk 2 above)
Comment (Rehaan1) [reply api/routes/search.js]: Will make the changes

**Review by rithikjain — APPROVED** [review]
LGTM. Can merge @BRO3886

**Pull request merged**